user-ops-indexer/user-ops-indexer-server/src/indexer.rs

```diff
@@ -1,33 +1,59 @@
 use crate::settings::Settings;
-use ethers::prelude::{Provider, Ws};
+use ethers::prelude::Provider;
 use sea_orm::DatabaseConnection;
+use std::sync::Arc;
+use user_ops_indexer_logic::indexer::{common_transport::CommonTransport, v06, v07};
 
 pub async fn run(
     settings: Settings,
     db_connection: DatabaseConnection,
 ) -> Result<(), anyhow::Error> {
     tracing::info!("connecting to rpc");
 
-    let ws_client = Ws::connect_with_reconnects(settings.indexer.rpc_url, 20).await?;
-    let client = Provider::new(ws_client);
+    let db_connection = Arc::new(db_connection);
+
+    let transport = CommonTransport::new(settings.indexer.rpc_url.clone()).await?;
+    let supports_subscriptions = matches!(transport, CommonTransport::Ws(_));
```

Comment:
I guess it's better to move this logic of "does support subscription" to CommonTransport then

like `let supports_subscriptions = transport.support_subscription()`